addons/stock_transport/models/stock_picking_batch.py

```diff
@@ -0,0 +1,112 @@
+        pass
+
+    @api.depends('vehicle_id', 'max_picking_weight', 'max_picking_volume')
+    def _compute_vehicle_details(self):
+        for record in self:
+            vehicle_details = []
+            if record.vehicle_id:
+                vehicle_details.append(record.vehicle_id.name)
+            vehicle_details.append(f"{round(record.max_picking_weight, 2)} Kg") if record.max_picking_weight else vehicle_details.append("0Kg")
+            vehicle_details.append(f"{round(record.max_picking_volume, 2)} m³") if record.max_picking_volume else vehicle_details.append("0m³")
+            record.vehicle_details = ', '.join(vehicle_details)
+
+    @api.depends('picking_ids.max_weight', 'vehicle_category_id.max_weight')
+    def _compute_used_weight_percentage(self):
+        for batch in self:
+            batch.max_picking_weight = sum(picking.max_weight for picking in batch.picking_ids)
+            batch_vehicle_max_weight = batch.vehicle_category_id.max_weight
+            used_weight_percentage = 100 * (batch.max_picking_weight / batch_vehicle_max_weight) if batch_vehicle_max_weight else 0.0
+            batch.used_weight_percentage = used_weight_percentage
+
+    @api.depends('picking_ids.max_volume', 'vehicle_category_id.max_volume')
+    def _compute_used_volume_percentage(self):
+        for batch in self:
+            batch.max_picking_volume = sum(picking.max_volume for picking in batch.picking_ids)
+            batch_vehicle_max_volume = batch.vehicle_category_id.max_volume
+            used_volume_percentage = 100 * (batch.max_picking_volume / batch_vehicle_max_volume) if batch_vehicle_max_volume else 0.0
+            batch.used_volume_percentage = used_volume_percentage
+
+    @api.depends('picking_type_id', 'max_picking_weight', 'max_picking_volume')
+    def _compute_picking_info(self):
+        for record in self:
+            picking_type = f"{record.picking_type_id.name}: " if record.picking_type_id else ''
+            max_picking_weight = f"{round(record.max_picking_weight, 2)}kg" if record.max_picking_weight else '0Kg'
+            max_picking_volume = f"{round(record.max_picking_volume, 2)}m³" if record.max_picking_volume else '0m³'
+            picking_info_str = f"{picking_type}{max_picking_weight}, {max_picking_volume}".rstrip(', ')
+            record.picking_info = picking_info_str
+
+    @api.depends('picking_ids.sequence')
+    def _compute_sorted_picking_ids(self):
+        for batch in self:
+            batch.sorted_picking_ids = batch.picking_ids.sorted(key=lambda p: p.sequence)
+
+    @api.depends('dock_id')
+    def _compute_move_ids(self):
+        super()._compute_move_ids()
+        if self.dock_id:
+            parent_path_id = [int(parent_id) for parent_id in self.dock_id.parent_path.split('/')[:-1]]
+            for line in self.move_line_ids:
+                if line.location_dest_id.id in parent_path_id:
+                    line.location_dest_id = self.dock_id
```

Comment:
If you use a dock, write the doc location on the picking in the batch and let them handle the stock.move.line

Reply:
Yeah, I agree with you but it’s a standard bug changing on picking it is handling the move line automatically. I have checked this in standard (by changing the location on picking it doesn’t change that location on the move line automatically. So to handle it I need to set on picking and move lines manually.

below is video link for reference-
https://www.awesomescreenshot.com/video/28565698?key=09164f5d2459cb8475f3e83a76c2da38